lib/experimental/fonts-library/class-wp-font-family.php

```diff
@@ -0,0 +1,443 @@
+	 */
+	private function download_asset( $src, $filename ) {
+		// Checks if the file to be downloaded has a font mime type.
+		if ( ! WP_Font_Family_Utils::has_font_mime_type( $filename ) ) {
+			return false;
+		}
+
+		// Include file with download_url() if function doesn't exist.
+		if ( ! function_exists( 'download_url' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		// Downloads the font asset or returns false.
+		$temp_file = download_url( $src );
+		if ( is_wp_error( $temp_file ) ) {
+			return false;
+		}
+
+		$overrides = array(
+			'action'                   => 'wp_handle_font_upload', // Arbitrary string to avoid the is_uploaded_file() check applied when using 'wp_handle_upload'.
+			'test_form'                => false, // We are not testing a form submission.
+			'test_type'                => false, // Seems like we can not test mime type for files that are not images. See this function docs: https://developer.wordpress.org/reference/functions/wp_check_filetype_and_ext/ .
+			'unique_filename_callback' => function () use ( $filename ) {
+				return $filename; }, // We want to keep the original filename.
+		);
+
+		$file = array(
+			'tmp_name' => $temp_file,
+			'name'     => $filename,
+		);
+
+		$handled_file = wp_handle_upload( $file, $overrides );
+
+		// Cleans the temp file.
+		@unlink( $temp_file );
+
+		if ( ! isset( $handled_file['url'] ) ) {
+			return false;
+		}
+
+		// Returns the relative path to the downloaded font asset to be used as font face src.
+		return $handled_file['url'];
+	}
+
+	/**
+	 * Move an uploaded font face asset from temp folder to the wp fonts directory.
+	 *
+	 * This is used when uploading local fonts.
+	 *
+	 * @param array $font_face Font face to download.
+	 * @param array $file Uploaded file.
+	 * @return array New font face with all assets downloaded and referenced in the font face definition
+	 */
+	private function move_font_face_asset( $font_face, $file ) {
+		$new_font_face = $font_face;
+		$filename      = WP_Font_Family_Utils::get_filename_from_font_face( $this->data['slug'], $font_face, $file['name'] );
+
+		// Remove the uploaded font asset reference from the font face definition because it is no longer needed.
+		unset( $new_font_face['uploaded_file'] );
+
+		// If the filename has not a font mime type, we don't move the file and return the font face definition without src to be ignored later.
+		if ( ! WP_Font_Family_Utils::has_font_mime_type( $filename ) ) {
+			return $new_font_face;
+		}
+
+		// Move the uploaded font asset from the temp folder to the wp fonts directory.
+		if ( ! function_exists( 'wp_handle_upload' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		$overrides = array(
+			'action'                   => 'wp_handle_font_upload', // Arbitrary string to avoid the is_uploaded_file() check applied when using 'wp_handle_upload'.
```

Comment:
See my earlier comment with suggestions regarding inline comments/closure `}` alignment.

Given that this is the same as the other usage with the exception of `$filename`, it may be worth abstracting this to a method such as `get_upload_overrides( $filename )` to avoid code repetition.